src/modules/lost/utilities.c

```diff
@@ -0,0 +1,592 @@
+/*
+ * lost module utility functions
+ *
+ * Copyright (C) 2019 Wolfgang Kampichler
+ * DEC112, FREQUENTIS AG
+ *
+ * This file is part of Kamailio, a free SIP server.
+ *
+ * Kamailio is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version
+ *
+ * Kamailio is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+/*!
+ * \file
+ * \brief Kamailio lost :: utilities
+ * \ingroup lost
+ * Module: \ref lost
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <time.h>
+
+#include <libxml/xmlmemory.h>
+#include <libxml/parser.h>
+#include "../../core/parser/msg_parser.h"
+#include "../../core/parser/parse_content.h"
+#include "../../core/parser/parse_uri.h"
+#include "../../core/parser/parse_from.h"
+#include "../../core/parser/parse_ppi_pai.h"
+#include "../../core/dprint.h"
+#include "../../core/mem/mem.h"
+#include "../../core/mem/shm_mem.h"
+
+#include "pidf.h"
+#include "utilities.h"
+
+/*
+ * lost_trim_content(dest, lgth)
+ * removes whitespace that my occur in a content of an xml element
+ */
+char *lost_trim_content(char *str, int *lgth)
+{
+	char *end;
+
+	while(isspace(*str))
+		str++;
+
+	if(*str == 0)
+		return NULL;
+
+	end = str + strlen(str) - 1;
+
+	while(end > str && isspace(*end))
+		end--;
+
+	*(end + 1) = '\0';
+
+	*lgth = (end + 1) - str;
+
+	return str;
+}
+
+/*
+ * lost_rand_str(dest, length)
+ * creates a random string used as temporary id in a findService request
+ */
+void lost_rand_str(char *dest, size_t lgth)
+{
+	size_t index;
+	char charset[] = "0123456789"
+					 "abcdefghijklmnopqrstuvwxyz"
+					 "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+	srand(time(NULL));
+	while(lgth-- > 0) {
+		index = (double)rand() / RAND_MAX * (sizeof charset - 1);
+		*dest++ = charset[index];
+	}
+	*dest = '\0';
+}
+
+/*
+ * lost_free_loc(ptr)
+ * freess a location object
+ */
+void lost_free_loc(p_loc_t ptr)
+{
+	pkg_free(ptr->identity);
+	pkg_free(ptr->urn);
+	pkg_free(ptr->longitude);
+	pkg_free(ptr->latitude);
+	pkg_free(ptr);
+}
+
+/*
+ * lost_new_loc(urn)
+ * creates a new location object in private memory and returns a pointer
+ */
+p_loc_t lost_new_loc(str rurn)

```

Comment:
Makes it sense to return a invalid pointer if the one or two of the allocations fail? You also could do a memset on an invalid pointer.